# HG changeset patch
# User Rik <rik@octave.org>
# Date 1549160484 28800
#      Sat Feb 02 18:21:24 2019 -0800
# Node ID fec69a3108337a7c86305ec7f96a424fa1694d02
# Parent  3c4ba50b20244bc1d34eb5b100ffab84ad815c77
__ezplot__.m: for ezplot3, draw "animate" plot just once.

* __ezplot__.m: If "animate" option given for ezplot3, draw plot using
comet3().  Otherwise, draw plot using plot3().

diff --git a/scripts/plot/draw/private/__ezplot__.m b/scripts/plot/draw/private/__ezplot__.m
--- a/scripts/plot/draw/private/__ezplot__.m
+++ b/scripts/plot/draw/private/__ezplot__.m
@@ -440,20 +440,20 @@ function [h, needusage] = __ezplot__ (pl
         if (! parametric)
           axis (hax, domain);
         else
           axis ("equal");
         endif
       endif
     elseif (isplot3)
       if (animate)
-        ## draw animation, then replace with true plot3
-        comet3 (hax, X, Y, Z, .05);
+        comet3 (hax, X, Y, Z);
+      else
+        h = feval (pltfunc, hax, X, Y, Z);
       endif
-      h = feval (pltfunc, hax, X, Y, Z);
       grid (hax, "on");
       zlabel (hax, "z");
     else  # mesh and surf plots
       h = feval (pltfunc, hax, X, Y, Z);
       ## FIXME: surf, mesh should really do a better job of setting zlim
       if (! parametric)
         axis (hax, domain);
       endif
